### PR TITLE
Slightly improve readme

### PR DESCRIPTION
**Pull request by blattms**
Make it clear what is in the repository. Might help people finding it via search engines.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+-------
+
+The library is distributed under the GNU General Public License,
+version 3 or later (GPLv3+).
+
+Requirements
+-----------
+
+- C++17 compatible compiler
+- CMake >= 3.10
```
Comment (bska) [inline README.md:22]: We **may** want to add "Posix compatibility" here since we do use Posix functions, notably `fnmatch`, in a number of locations.

Comment (blattms) [reply README.md:22]: Lets wait for #2925

Comment (bska) [reply README.md:22]: > Let's wait for #2925

PR #2925 definitely helps, but doesn't fully resolve the issue.  We use `fileno()` and `isatty()` from `<unistd.h>` as part of our [logging](https://github.com/OPM/opm-common/blob/24e58e36aab0edfc1094b0725c1300e017a8f3b7/src/opm/common/OpmLog/OpmLog.cpp#L43) system (colourised output).  We could **probably** push that functionality into libfmt (using function `fmt::print()`), but that would require quite a bit of work in our logger, and also prevent fully switching to `std::format()` if we upgrade to C++20 at a later date.  We also have a few unit tests that use `<unistd.h>` and even [`<sys/*.h>`](https://github.com/OPM/opm-common/blob/24e58e36aab0edfc1094b0725c1300e017a8f3b7/tests/parser/ParseContext_EXIT1.cpp#L3-L4), but these can obviously be disabled if a user does not want them.

Comment (bska) [reply README.md:22]: One more case that requires Posix compatibility: We have a number of example and testing utilities, e.g.,

* [`convertECL`](https://github.com/OPM/opm-common/blob/089b6bff42fffa4b3291591f5779b521d9717b52/test_util/convertECL.cpp#L133)
* [`compareECL`](https://github.com/OPM/opm-common/blob/089b6bff42fffa4b3291591f5779b521d9717b52/test_util/compareECL.cpp#L99)
* [`opmpack`](https://github.com/OPM/opm-common/blob/089b6bff42fffa4b3291591f5779b521d9717b52/examples/opmpack.cpp#L110)

that all use `getopt()` from `<unistd.h>` (or `<getopt.h>`).

Comment (blattms) [reply README.md:22]: added

**Review by bska — APPROVED** [review]
Thanks for the updates.  I'll merge this into master now.

**Pull request merged**